acme/http_test.go

```diff
@@ -212,6 +212,150 @@ func TestRetryBackoffArgs(t *testing.T) {
+			expectedErr: &Error{
+				StatusCode: http.StatusTooManyRequests,
+			},
+		},
+		{
+			desc:                   "does not retry get request that returns response code that is not retriable in default isRetriable configuration",
+			requestMethodType:      http.MethodGet,
+			responseHttpStatusCode: http.StatusNotFound,
+			expectedRetryCount:     0,
+			expectedErr: &Error{
+				StatusCode: http.StatusNotFound,
+			},
+		},
+		{
+			desc:                   "retries post request that returns response code that is retriable in the client.ShouldRetry configuration",
+			requestMethodType:      http.MethodPost,
+			responseHttpStatusCode: http.StatusInternalServerError,
+			isRetriable: func(resp *http.Response) bool {
+				return resp.StatusCode >= 500
+			},
+			expectedRetryCount: 1,
+			expectedErr: &Error{
+				StatusCode: http.StatusInternalServerError,
+			},
+		},
+		{
+			desc:                   "does not retry post request that returns response code that is not retriable in the client.ShouldRetry configuration",
+			requestMethodType:      http.MethodPost,
+			responseHttpStatusCode: http.StatusTooManyRequests,
+			isRetriable: func(resp *http.Response) bool {
+				return false
+			},
+			expectedRetryCount: 0,
+			expectedErr: &Error{
+				StatusCode: http.StatusTooManyRequests,
+			},
+		},
+		{
+			desc:                   "retries get request that returns response code that is retriable in the client.ShouldRetry configuration",
+			requestMethodType:      http.MethodGet,
+			responseHttpStatusCode: http.StatusTooManyRequests,
+			isRetriable: func(resp *http.Response) bool {
+				return resp.StatusCode == http.StatusTooManyRequests
+			},
+			expectedRetryCount: 1,
+			expectedErr: &Error{
+				StatusCode: http.StatusTooManyRequests,
+			},
+		},
+		{
+			desc:                   "does not retry get request that returns response code that is not retriable in the client.ShouldRetry configuration",
+			requestMethodType:      http.MethodGet,
+			responseHttpStatusCode: http.StatusTooManyRequests,
+			isRetriable: func(resp *http.Response) bool {
+				return resp.StatusCode <= 399 || resp.StatusCode >= 500
+			},
+			expectedRetryCount: 0,
+			expectedErr: &Error{
+				StatusCode: http.StatusTooManyRequests,
+			},
+		},
+	}
+
+	for _, test := range tt {
+		t.Run(test.desc, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(test.responseHttpStatusCode)
+			}))
+			defer ts.Close()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			var retryNr int
+
+			client := &Client{
+				Key: testKey,
+				RetryBackoff: func(n int, r *http.Request, res *http.Response) time.Duration {
+					retryNr++
+					if retryNr == 1 {
+						cancel()
+					}
+					return time.Millisecond
+				},
+				dir: &Directory{AuthzURL: ts.URL},
+			}
+
+			if test.isRetriable != nil {
+				client.ShouldRetry = test.isRetriable
```

Comment:
I think you can always assign here yeah? The constructor will take care of if it's `nil`